### ISSUE TITLE
Suggestion: Support callables as parameter for url_for

### ISSUE THREAD

**Issue by csachs** [pallets/flask#2189]
I'd suggest enhancing the `url_for` function to accept callables as well, i.e. the function name unquoted; as this would help e.g. with code refactoring tools and auto completion.

It can easily be implemented with a few lines; if such a functionality would be desirable, I can prepare a PR with tests, etc.

Best regards, Christian

**Comment by ThiefMaster** [comment]
I don't think this makes much sense - it'd just make it more likely for people to end up in import hell.

FYI, the endpoint name doesn't even need to match the function name - this is just a default that works in many cases. Also, it is very common to use blueprints where you'd actually generate a url for `someblueprint.someendpoint`. Again, using the actual functions there would likely result in import hell.

**Comment by csachs** [comment]
I'm monkey patching it in my projects and prefer it as cleaner, compared to cluttering strings everywhere, but OK…

**Comment by untitaker** [comment]
@ThiefMaster gave you an argument why such behavior would be technically incorrect. You're right that this would help with tab-completion, but I don't think that's a good reason to mangle an API. If you want to use it in your own projects, that's totally fine. I don't think you need to monkeypatch anything though, because you can always just import your new `url_for` function and use that, no?

**Comment by davidism** [comment]
Also, the same view function can be registered with multiple endpoint names. Which one should it reverse to in such a case? When using the endpoint names, it's unambiguous.

**Comment by csachs** [comment]
I was not aware of an completely independent endpoint mapping distinct from the routing; so far I've only used routing decorators.
Different blueprints could probably be accounted for by e.g. passing them as parameter as well…
The possible ambiguity should not be much of a problem if the resulting URL leads to identical function calls upon request, and if someone wants to prevent ambiguity, they could always use the string variant, which I did not suggest to abandon.
